pkg/operation/botanist/namespaces.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/botanist/systemcomponents/namespaces"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// DeploySeedNamespace creates a namespace in the Seed cluster which is used to deploy all the control plane
+// components for the Shoot cluster. Moreover, the cloud provider configuration and all the secrets will be
+// stored as ConfigMaps/Secrets.
+func (b *Botanist) DeploySeedNamespace(ctx context.Context) error {
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: b.Shoot.SeedNamespace,
+		},
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, b.K8sSeedClient.Client(), namespace, func() error {
+		namespace.Annotations = map[string]string{
+			v1beta1constants.DeprecatedShootUID: string(b.Shoot.Info.Status.UID),
+		}
+		namespace.Labels = map[string]string{
+			v1beta1constants.GardenRole:              v1beta1constants.GardenRoleShoot,
+			v1beta1constants.LabelSeedProvider:       b.Seed.Info.Spec.Provider.Type,
+			v1beta1constants.LabelShootProvider:      b.Shoot.Info.Spec.Provider.Type,
+			v1beta1constants.LabelNetworkingProvider: b.Shoot.Info.Spec.Networking.Type,
+			v1beta1constants.LabelBackupProvider:     b.Seed.Info.Spec.Provider.Type,
+		}
+
+		if b.Seed.Info.Spec.Backup != nil {
+			namespace.Labels[v1beta1constants.LabelBackupProvider] = b.Seed.Info.Spec.Backup.Provider
+		}
+
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	b.SeedNamespaceObject = namespace
+	return nil
+}
+
+// DeleteSeedNamespace deletes the namespace in the Seed cluster which holds the control plane components. The built-in
+// garbage collection in Kubernetes will automatically delete all resources which belong to this namespace. This
+// comprises volumes and load balancers as well.
```

Comment:
```suggestion
// comprises volumes claims and load balancers as well.
```

Reply:
I'll fix it in another PR, I don't want to trigger the whole CI+approval flow now again. Thanks.